docs/bugs_and_glitches.md

```diff
@@ -256,7 +256,18 @@ This is a bug with `CheckPlayerHasUsableMoves` in [battle/core.asm](/battle/core
 
 This bug affects Attract, Curse, Foresight, Mean Look, Mimic, Nightmare, Spider Web, Transform, and stat-lowering effects of moves like String Shot or Bubble during the semi-invulnerable turn of Fly or Dig.
 
-*To do:* Identify specific code causing this bug and fix it.
```

Comment:
This bug affects virtually all moves and the problem seems to come from https://github.com/roukaour/pokecrystal/blob/2fa1e97a/battle/effect_commands.asm#L1823
More precisely, it seems to be that the line after the caller (https://github.com/roukaour/pokecrystal/blob/2fa1e97a/battle/effect_commands.asm#L1717) should've been ret z instead.

Reply:
`.LockOn` returns `nz` if the move will definitely hit, `z` if it might still miss. `BattleCommand_CheckHit` returns early if the move will definitely hit; if the move misses, it has to set `[AttackMissed]` to 1. So `call .LockOn; ret nz` looks correct.

Reply:
in any case, `.lockOn` returns nz when the target is during Fly or Dig, causing BattleCommand_CheckHit to return early and ensure the hit.

Reply:
I'm just not sure L1717 `ret nz` is the one to change. Wouldn't that make moves always hit when they're *not* locked on? The control flow within `.LockOn` looks complex enough that it might be giving the wrong result in some cases.

Reply:
Yeah, my bad. That change would lead to non-lock on scenarios being treated as locked on. I'm not sure about potential side effects but on the other hand replacing the ``ret nz``, with ``jp nz, .Miss`` could do the trick (although this does seem to be ``.FlyDigMoves`` work, so ``.LockOn`` itself may need a rework. Basically, you want to catch the cases where the target is locked on but not during Fly/Dig, in order to ensure the hit in those. That said, it would make more sense if ``.FlyDigMoves`` was called before ``.LockOn`` so that it doesn't have to handle more complex scenarios than necessary.